geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -77,18 +77,15 @@ public RedisSet() {}
 
   public static int smove(RedisKey sourceKey, RedisKey destKey, byte[] member,
       RegionProvider regionProvider) {
-    Region<RedisKey, RedisData> region = regionProvider.getDataRegion();
-
     RedisSet source = regionProvider.getTypedRedisData(REDIS_SET, sourceKey, false);
     RedisSet destination = regionProvider.getTypedRedisData(REDIS_SET, destKey, false);
-    if (source.members.isEmpty() || !source.members.contains(member)) {
+    if (source == NULL_REDIS_SET || !source.members.contains(member)) {
       return 0;
     }
-
-    List<byte[]> movedMember = new ArrayList<>();
-    movedMember.add(member);
-    source.srem(movedMember, region, sourceKey);
-    destination.sadd(movedMember, region, destKey);
+    List<byte[]> memberList = new ArrayList<>();
+    memberList.add(member);
+    source.srem(memberList, regionProvider, sourceKey);
```

Comment:
srem and sadd actually modifies "memberList" but they do not make clear how they modify it. So I was worried that this code might get in trouble in reusing "memberList" for the sadd call. But it turns out it is okay because srem only removes members from the list that it did not remove and sadd only removes members from the list that it did not add. I think it might be helpful to spell this out in the javadoc comment on these two methods (something like NOTE this list will be modified to only contains members removed/added).

Reply:
What if I refactored the sadd and srem methods to not modify the list? That sort of side effect feels a bit risky to me to begin with, and it's not necessary if we just iterate the list and add to the delta as we go rather than creating the delta after doing all the adds/removes. The only real performance difference is that if we do an SADD or SREM that ends up not adding or removing anything, now we would be creating a single `DeltaInfo` that we weren't previously, which isn't too expensive.